base/shell/explorer/explorer.cpp

```diff
@@ -87,6 +87,37 @@ HideMinimizedWindows(IN BOOL bHide)
 }
 #endif
 
+static BOOL
+IsExplorerSystemShell()
+{
+    HKEY hKeyWinlogon;
+	WCHAR szShell[256];
+    WCHAR szExplorer[] = L"explorer";
```

Comment:
If our explorer executable gets renamed (like, "rosplorer", and the value in the registry value adjusted as well), your code wouldn't work.

Reply:
Have you considered getting the full path of Explorer’s HINSTANCE and comparing it with the value in the Registry? This way you can call the executable whatever you want and it will still work.

Reply:
You've to be careful, because the registry value can contain only "explorer.exe" (for example), but the full module path be "c:\reactos\system32\explorer.exe", or if you try to execute a copy of explorer.exe that is in a different directory, it could be completely different.

Reply:
Yes, it needs PathFindFileName

Reply:
Fixed in latest commit